src/Raven.Server/Documents/Sharding/Handlers/Processors/Collections/ShardedCollectionsHandlerProcessorForGetCollectionDocuments.cs

```diff
@@ -0,0 +1,124 @@
+using Raven.Server.Documents.Sharding.Streaming;
+using Raven.Server.Json;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Collections
+{
+    internal class ShardedCollectionsHandlerProcessorForGetCollectionDocuments : AbstractCollectionsHandlerProcessorForGetCollectionDocuments<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedCollectionsHandlerProcessorForGetCollectionDocuments([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler)
+        {
+        }
+
+        protected override async ValueTask GetCollectionDocumentsAndWriteAsync(TransactionOperationContext context, string name, int _, int __, CancellationToken token)
+        {
+            throw new NotSupportedInShardingException($"Fetching documents by collection not supported in sharding.");
+
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal,
+                "Need to have ShardedStreamDocumentsOperation fetch by collection name");
+
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Arek, DevelopmentHelper.Severity.Normal,
+                "See `null` passed as etag to above new ShardedCollectionPreviewOperation()");
+
+            var continuationToken = RequestHandler.ContinuationTokens.GetOrCreateContinuationToken(context);
+
+            var op = new ShardedStreamDocumentsOperation(HttpContext, null, continuationToken);
+            var results = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op);
+            using var streams = await results.Result.InitializeAsync(RequestHandler.DatabaseContext, HttpContext.RequestAborted);
+            var documents = GetDocuments(streams, continuationToken);
+
+            long numberOfResults;
+            long totalDocumentsSizeInBytes;
+
+            await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+            {
+                writer.WriteStartObject();
+                writer.WritePropertyName("Results");
+                (numberOfResults, totalDocumentsSizeInBytes) = await writer.WriteDocumentsAsync(context, documents, metadataOnly: false, token);
+
+                if (continuationToken != null)
+                {
+                    writer.WriteComma();
+                    writer.WriteContinuationToken(context, continuationToken);
+                }
+                writer.WriteEndObject();
+            }
+
+            //AddPagingPerformanceHint(PagingOperationType.Documents, "Collection", HttpContext.Request.QueryString.Value, numberOfResults, pageSize, sw.ElapsedMilliseconds, totalDocumentsSizeInBytes);
+            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal, "Handle sharded AddPagingPerformanceHint");
```

Comment:
probably better to return numberOfResults and totalDocumentsSizeInBytes from this method and handle performance hint externally in the abstract?